proposals/lip-0041.md

```diff
@@ -250,43 +555,101 @@ The Auth module does not execute any logic during the genesis state finalization
 
 #### Transaction Verification
 
-As part of the verification of a transaction `trs`, the following checks are applied. If the checks fail the transaction is invalid and the whole block is discarded.
+As part of the verification of a transaction `trs`, we do the following:
 
-* Check that the transaction nonce is equal to the account nonce by returning `isValidNonce(trs)`.
+* Check that the transaction nonce is equal to the account nonce or to 0 if the account does not exist. If the checks fail the transaction is invalid and the whole block is discarded.
+Notice that a transaction in the transaction pool with a nonce greater than the account nonce is considered "pending" rather than invalid, since it could become valid in the future.
+* Check the signatures property of the transaction [as explained in LIP 0017](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0017.md#signatures-replaces-signature) using `authAccount(address)`, where `address` is the address corresponding to `trs.senderPublicKey`.
 
 ```python
-isValidNonce(trs):
-  let senderAddress be the address corresponding to trs.senderPublicKey
-  if there is no entry in the in the auth data substore with storeKey == senderAddress:
-    return trs.nonce == 0
-  return trs.nonce == authAccount(senderAddress).nonce
+def verifyTransaction(trs: Transaction) -> None:
+    verifyNonce(trs)
+    verifySignature(trs)
 ```
+ 
+We define the following auxiliary function to verify the nonce of a transaction. 
 
-Notice that a transaction in the transaction pool with a nonce greater than the account nonce is considered "pending" rather than invalid, since it could become valid in the future.
+```python
+def verifyNonce(trs: Transaction) -> None:
+    senderAddress = 20-byte address derived from trs.senderPublicKey
+    if there is no entry in the in the auth data substore with storeKey == senderAddress:
+        expectedNonce = 0
+    else:
+        expectedNonce = authAccount(senderAddress).nonce
+    
+    if trs.nonce != expectedNonce:
+        raise Exception(f'Invalid nonce property. Expected {expectedNonce}, got {trs.nonce}.')
+```
 
-* Check the signatures property of the transaction [as explained in LIP 0017](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0017.md#signatures-replaces-signature) using `authAccount(address)`, where `address` is the address corresponding to `trs.senderPublicKey`.
+We define the following auxiliary function to verify the validity of a signature. These specifications supersede those given in [LIP 0017](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0017.md#signatures-replaces-signature). The function `verifyMessageSig` is defined in [LIP 0062](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0062.md#specification) and `networkIdentifier` is the network identifier of the chain.
+
+```python
+def verifySignature(trs: Transaction) -> None:
```

Comment:
I think tabs and spaces are mixed here. At least on GitHub the indentations are off sometimes.